### DISCUSSION TITLE
Keep track of element index filters

### DISCUSSION THREAD

**Discussion by pixelmachine** [craftcms/cms#12590] [Ideas] [answered]
### What happened?

The recently added element index filters are amazing and powerful but pretty fragile. Once you go through the multiple steps to create a filter there's no way to keep that filter and it's immediately lost on any navigation.

### Steps to reproduce

1. Make a filter
2. Click an entry
3. Click 'Entries' and start from scratch

### Expected behavior

User can save the search to be reused along with other searches, or the search is temporarily stored (cookie or whatever) until another filter is setup or the active filter is cleared.

### Actual behavior

Filter is always lost

### Use case

I have a site that has thousands of entries related to many categories and makes sense to be able to filter the list to see a specific category's entries and work on many of them. Re-adding a filter each time is painful and setting up custom sources for each category is not something that can be done on the fly and would be a bad UI if it could as the list would get long.

### Notes

Brandon mentioned [on Discord](https://discord.com/channels/456442477667418113/456442477667418115/964515590775668766) that there is some intent to do this but I couldn't find any other mention on Github.

I think it's kind of related to a [discussion I added](https://github.com/craftcms/cms/discussions/11781) in that this would be a sort of workaround in some respects.

### Craft CMS version

4.3.6

**Comment by brandonkelly** [comment]
Converted this to a discussion because it’s currently working as intended. Agree that it would be good to keep track of the active filters, though. Maybe even from a query string so the results could be shared with others (similar to the current search term – #8942).

**Reply by pixelmachine** [reply]
Yep fair enough. 

I think really there are two requests here:

1. Maintain the active search because it's frustrating to keep setting up the same query over and over when running through entries.  I think this one is the key one as you said. 

2. Save searches a bit like custom element sources so they can be reused on the fly

I was originally going to suggest query string (that's what I suggested in my linked discussion) but I wondered if that might still be a bit fragile or frustrating if you accidentally navigate away?

**Comment by markhuot** [comment]
The idea of "remembered" and "saved" filters is a huge selling point of a few "enterprise-y" CMSs I've come across recently. One that really stuck with an editorial team had the flow of:

1. create a filter on the entry listing page, get a result set of say, 100+ entries
2. click in to the first entry to make changes

From there you have two options,

1. click _back_ in the browser to go back to the search (stored in query string) and then click on the second result
2. on the edit screen a "ribbon" appears showing the other entries from the result set (if it was filtered) so you could just page through results right from the edit screen. This flow was immensely helpful, because you could queue up a list of entries you're working on based on your filter and then just work your way through.

All that is to say, yes, saved filters would be an amazing feature!

**Comment by thupsi** [comment]
There was an earlier dicussion that was relative to this and Brandon said that savable filters are planned: https://github.com/craftcms/cms/discussions/11215#discussion-4070110

No idea about a timeframe though. FWIW I also would love to see this implemented!

**Comment by brandonkelly** [comment, marked as answer]
Craft 4.5.0 is out with an improvement related to this: filters are now serialized and added to the query string, so they will persist when you navigate to an entry and then press the Back button, or if you share the index page URL with a colleague.

**Comment by outline4** [comment]
Hi, 

I just had a very confused client telling me that the filter is lost each time. 

I had to advise him to command-save the entry and then use the back button of the browser to return to the filtered elements. 
That is counter intuitive. Using the back button of the browser in a CMS context is not great. 

The more optimal workflow could be that if you save an entry you'd return to the filtered list. 
If you click on entries -> you also return to the filterd list. 

In fact the element index should always keep the filter unless you decide to delete it (on a user basis).

**Comment by janhenckens** [comment]
Extra vote for this - I'm building out a site where the client uses Craft as a mini-crm for their users/contacts and I'd love to have the option for them to save the filters they created - is this something that is on the roadmap @brandonkelly ?
